cpp/src/lance/io/writer.h

```diff
@@ -53,8 +53,12 @@ class FileWriter final : public ::arrow::dataset::FileWriter {
 
   ::arrow::Status WriteArray(const std::shared_ptr<format::Field>& field,
                              const std::shared_ptr<::arrow::Array>& arr);
-  ::arrow::Status WritePrimitiveArray(const std::shared_ptr<format::Field>& field,
-                                      const std::shared_ptr<::arrow::Array>& arr);
+
+  /// Write fixed size array, including primitive arrays, fixed size binary array,
+  /// and fixed size list array.
+  ::arrow::Status WriteFixedSizedArray(const std::shared_ptr<format::Field>& field,
```

Comment:
it's a little confusing that this name makes you think it's dealing with fixed sized list / fixed sized binary only. If it's not too much a hassle, can we consider renaming this to something more generic like "WriteFixedStrideArray" or something like that?

Reply:
Sure , can do

Reply:
yeah that sounds fine